tracks/python/exercises/isogram/mentoring.md

```diff
@@ -0,0 +1,112 @@
+### Problem and challenges
+
+The solution looks for duplicate letters.
+
+### Reasonable solutions
+	
+This version searches for a second copy of each letter in turn
+	
+	```python
+        def is_isogram(string):
+
+        string = string.lower()
+
+        for i in range(len(string)):
+            ch = string[i]
+
+            if ch.isalpha():
+                if (s.find(ch) > -1):
+                    return False
+
+        return True
+	```
+	
+This version needs the index to construct the substring of remaining characters.
+
+#### Sorting the string
+    
+Another approach is to sort the input, and walk down it looking for
+duplicates.  However, Python provides a number of useful tools that allow
+alternative approaches.  
+
+#### Trimming the string
+
+If the approach is to remove non-letters, it is simpler to look for things that
+belong (letters) than things that do not belong - a much harder list to define.
+
+    It is common to see students constructing strings by addition with a
+    fragment like this:
+
+	```python
+        result = ""
+        for ch in string:
+            if ch.isalpha()
+                result = result + ch
+	```
+
+Building strings is slow in Python, and it is much better to create a list and join() it.
```

Comment:
I think you mean String concatenation is slow in Python (or building a string via concatenation).